syntax = "proto3";

package kraken;

message KrakenRequest {
    string collector_name = 1;
    string content_type = 2;
    string metadata = 3;
    bytes payload = 4;
}

message KrakenResponse {
    string collector_name = 1;
    string content_type = 2;
    string metadata = 3;
    bytes payload = 4;
}

service KrakenService {
    rpc ProcessKrakenRequest (KrakenRequest) returns (KrakenResponse);
}